kuma/javascript/src/banners.jsx

```diff
@@ -211,6 +245,18 @@ export default function Banners(props: BannersProps) {
         let bannersList = props.banners || defaultBanners;
         for (const banner of bannersList) {
             if (userData.waffle.flags[banner.id]) {
+                // if this banner is to be shown to authenticated users only,
+                // and the current user is not authenticated, return `null`
+                if (banner.authenticated && !userData.isAuthenticated) {
+                    return null;
+                }
+
+                // special case for the subscriptions banner as we need access
+                // to the username as it forms part of the `url`
+                if (banner.id === 'subscription_banner' && userData.username) {
+                    banner.url = `/${locale}/profiles/${userData.username}/edit#subscription`;
```

Comment:
This needs to link to the Wiki.

Reply:
So, something like:

```
const wikiURL = "https://wiki.developer.mozilla.org";
...
banner.url = `${wikiURL}/${locale}/profiles/${userData.username}/edit#subscription`;
```

Reply:
@schalkneethling Is `win.mdn.wikiSiteUrl` available from the React pages?

Reply:
Let me give that a go. I was unsure of which portions of this happen on the server and which parts happen on the client. If on the server we will not but, the call to get `userData` happens inside a `useEffect` so we might be safe here.

Reply:
> @schalkneethling Is `win.mdn.wikiSiteUrl` available from the React pages?

It works o/\o

Reply:
Be careful there. Banners are server-side rendered. What's happening here is that they are *rendered* twice. Once on the server and once in the client. Actually, if you're logged in it's twice on the client. I.e. we're failing to react hydrate. This might have some performance degradations but if so very little. After all, in the server it rendered nothing and in the client it renders something. This is the same we're doing with the "Sign in" link. So the DOM, that the virtual DOM needs to *mutate*, is possibly a harmless (and necessary!) mutation. 
What I'm not entirely sure of is what React does when it freaks out during hydration. If it notices that the DOM doesn't match what the virtual DOM ought to be, does it trigger a full new rendering? I know, in development, it can lead to `console.warn` at most. 

Just be careful when using globals that can never exist in SSR. 

I think it's overkill right now but I think we should never do banners in server-side because of the nature of them potentially being 100% client-side. Any banner that depends on walffe FLAGs (not switches) or user ought to be started from a `useEffect` instead of the plain rendering path.